config/config.go

```diff
@@ -498,43 +486,45 @@ func (c Cluster) Config() (*Config, error) {
 		config.VPCRef = fmt.Sprintf("%q", config.VPCID)
 	}
 
+	//Assume Internet Gateway does not exist, reference by logical name
+	config.InternetGatewayRef = fmt.Sprintf(`{ "Ref" : %q }`, config.InternetGatewayLogicalName)
+	if config.InternetGatewayID != "" {
+		//This means this Internet Gateway already exists, and we can reference it directly by ID
+		config.InternetGatewayRef = fmt.Sprintf("%q", config.InternetGatewayID)
+	}
+
 	config.EtcdInstances = make([]etcdInstance, config.EtcdCount)
 	var etcdEndpoints, etcdInitialCluster bytes.Buffer
 
-	// Reset lastAllocatedAddr or we'll end up returning different cluster config w/ inconsistent static private ips
-	// for each time we call this function `cluster.Config()`
-	for _, subnet := range config.Subnets {
-		subnet.lastAllocatedAddr = nil
-	}
-
+	var lastAllocatedAddr = make(map[*model.Subnet]*net.IP)
 	for etcdIndex := 0; etcdIndex < config.EtcdCount; etcdIndex++ {
 
 		//Round-robbin etcd instances across all available subnets
 		subnetIndex := etcdIndex % len(config.Subnets)
 		subnet := config.Subnets[subnetIndex]
-		if config.EtcdPrivateSubnet {
-			subnet = config.EtcdSubnets[subnetIndex]
+		if config.Etcd.TopologyPrivate() {
```

Comment:
I like this way of automatically recognizing which topology an user meant to use 👍